iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -274,7 +278,18 @@ UnixDomainSocket::timedReceive(const units::Duration& timeout) const noexcept
     }
 
     struct timeval tv = timeout;
-    auto setsockoptCall = posixCall(setsockopt)(m_sockfd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv))
+#if defined(__APPLE__)
+    if (tv.tv_sec != 0 || tv.tv_usec != 0)
+    {
+        std::cerr
+            << "socket: \"" << m_name
+            << "\", timedReceive with a timeout != 0 is not supported on MacOS. timedReceive will behave like receive "
+               "instead."
+            << std::endl;
+    }
+#endif
```

Comment:
You are completely right, but this would be a more complex task since the timeout is not directly provided to the function, it is set in a call before via `setsockopt`.
I created an issue for this: https://github.com/eclipse-iceoryx/iceoryx/issues/823

Reply:
can this warning be printed in the `iox_setsockopt` implementation in the mac platform?

Reply:
That was my idea. The warning stays but is then printed via platforms.